flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/io/StreamTwoInputSelectableProcessor.java

```diff
@@ -286,64 +316,33 @@ private void checkAndSetAvailable(int inputIndex) {
 		}
 	}
 
-	/**
-	 * @return false if both of the inputs are finished, true otherwise.
-	 */
-	private boolean waitForAvailableInput(InputSelection inputSelection)
-		throws ExecutionException, InterruptedException, IOException {
-
+	private CompletableFuture<?> waitForAvailableInput(InputSelection inputSelection) {
```

Comment:
What about renaming all old `StreamTwoInputSelectable.waitFor...Input(...)` methods?
Now they are not actually waiting, but return "availability" future.
For example, they can be named as `getInputAvailable[Future](...), getOneInputAvailable[Future](...)`.